add compatibility with docker on mac

id opened the pull request:
i.e. make it possible to test brod as is on both mac and linux

zmstone commented:
the hostname for zookeeper is from `--link` option in docker run command for kafka container.
how does 'localhost' work ?

zmstone commented:
ok, I see the pausenet magic in your PR for brod.
perhaps parameterise the zookeeper hostname instead of hard-code it to 'localhost'.
also, please update README

id commented:
Updated README, added docker-compose instead of docker commands.
As for zookeeper hostname, is it really important? It was hardcoded to `zookeeper` before.

id commented:
Tested on OSX and Linux.

zmstone commented on docker-entrypoint.sh:
```diff
@@ -27,15 +27,16 @@ fi
 ipwithnetmask="$(ip -f inet addr show dev eth0 | awk '/inet / { print $2 }')"
 ipaddress="${ipwithnetmask%/*}"
 
+[ -z "$ADVERTISED_HOSTNAME" ] && ADVERTISED_HOSTNAME="localhost"
```
please default it to `ipaddress` so it's backward compatible.
(and you have this set explicitly in the compose file, which is good)

zmstone commented on docker-entrypoint.sh:
```diff
@@ -55,15 +56,15 @@ fi
 wait_for_kafka() {
   echo '### waiting for kafka to be ready'
-  if ! kafka-topics.sh --zookeeper zookeeper --list >/dev/null 2>&1; then
+  if ! kafka-topics.sh --zookeeper localhost --list >/dev/null 2>&1; then
```
and parameterise zookeeper host too in docker's env.

zmstone commented on server.properties:
```diff
@@ -114,7 +114,7 @@ log.retention.check.interval.ms=300000
 # root directory for all kafka znodes.
 ### changed for brod
-zookeeper.connect=zookeeper:2181
+zookeeper.connect=localhost:2181
```
then a `sed` is needed to replace it with the parameterised zookeeper hostname.
